src/dbt_osmosis/dbt_templater/templater.py

```diff
@@ -69,136 +60,78 @@ def process(self, *, fname: str, in_str=None, config, **kwargs):
-                if the_one:
-                    # Yes. Capture the important arguments and create
-                    # a make_template() function.
-                    env = args[0]
-                    globals = args[2] if len(args) >= 3 else kwargs["globals"]
-
-                    def make_template(in_str):
-                        env.add_extension(SnapshotExtension)
-                        return env.from_string(in_str, globals=globals)
-
-                    local.make_template = make_template
-
-        return old_from_string(*args, **kwargs)
-
-    def _unsafe_process(self, fname, in_str, config=None):
-        osmosis_dbt_project = self.dbt_project_container().get_project_by_root_dir(
+    def _unsafe_process(self, fname: Optional[str], in_str: str, config: FluffConfig = None):
+        # Get project
+        osmosis_dbt_project = self.dbt_project_container.get_project_by_root_dir(
+            # from .sqlfluff templater project_dir
             config.get_section((self.templater_selector, self.name, "project_dir"))
         )
-        local.make_template = None
-        try:
-            if fname:
-                node = self._find_node(osmosis_dbt_project, fname)
-                local.target_sql = Path(
-                    os.path.relpath(fname, start=osmosis_dbt_project.args.project_dir)
-                )
-                compiled_node = osmosis_dbt_project.compile_node(node)
-            else:
-                local.target_sql = in_str
-                # TRICKY: Use __wrapped__ to bypass the cache. We *must*
-                # recompile each time, because that's how we get the
-                # make_template() function.
-                compiled_node = osmosis_dbt_project.compile_sql.__wrapped__(
-                    osmosis_dbt_project, in_str
-                )
-        except Exception as err:
-            raise SQLFluffSkipFile(  # pragma: no cover
-                f"Skipped file {fname} because dbt raised a fatal "
-                f"exception during compilation: {err!s}"
-            ) from err
-        finally:
-            local.target_sql = None
-
-        if compiled_node.injected_sql:
-            # If injected SQL is present, it contains a better picture
-            # of what will actually hit the database (e.g. with tests).
-            # However it's not always present.
-            compiled_sql = compiled_node.injected_sql
-        else:
-            compiled_sql = compiled_node.compiled_sql
-
-        raw_sql = compiled_node.raw_sql
 
+        # Use path if valid, prioritize it as the in_str
+        fpath = Path(fname)
+        if fpath.exists() and not in_str:
+            in_str = fpath.read_text()
+
+        # Generate node
+        retry = 3
+        while retry > 0:
+            # In massive parallel requests, referring to higher loads than we will ever see in all likelihood,
+            # this cheap retry increases reliability to ~100% in test cases of up to 50 clients across 1K reqs
```

Comment:
When it fails and needs to retry, what errors were you seeing?

Reply:
It's a duplicate node name failure. Even using a uuidv4 I found it can happen. This is accounting for a real edge case but I feel the code is safer by virtue of having it than it is by not. 

On a deeper level for clarities sake, the get server node method actually adds a node to the manifest, the same manifest which we can access from multiple threads. So we make this operation atomic. I could probably hide it inside DbProject as a compile_node_safe as I think there is one other place I do this same thing. 


These failures only occur when you reach a certain level of concurrency tested on a pretty fast M1 chip.